api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirMedicationRequestDaoImpl.java

```diff
@@ -91,6 +95,9 @@ protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams
 					entry.getValue().forEach(d -> handleMedicationReference("d", (ReferenceAndListParam) d.getParam())
 					        .ifPresent(c -> criteria.createAlias("drug", "d").add(c)));
 					break;
+				case FhirConstants.STATUS_SEARCH_HANDLER:
```

Comment:
The "status_search_handler" is defined for the Task resource... I wasn't really sure if it made sense to repurpose it here, or if it makes more sense to create a new search handler... I'd need to understand better the concept behind a search handler.  Thoughts @ibacher ?

Reply:
I think it's fine to re-use it... If the string changes from `status`, presumably we'd use a different constant name.